internal/querynode/shard_cluster.go

```diff
@@ -611,15 +612,23 @@ func (sc *ShardCluster) HandoffSegments(info *querypb.SegmentChangeInfo) error {
 
 // appendHandoff adds the change info into pending list and returns the token.
 func (sc *ShardCluster) applySegmentChange(info *querypb.SegmentChangeInfo, onlineSegmentIDs []UniqueID) int64 {
+
+	// the suspects growing segment ids
+	// first all online segment shall be tried, for flush-handoff only puts segment in onlineSegments
+	// and we need to try all offlineSegments in case flush-compact-handoff case
+	possibleGrowingToRemove := make([]UniqueID, 0, len(info.OfflineSegments)+len(onlineSegmentIDs))
+	offlineSegments := typeutil.UniqueSet{} // map stores offline segment id for quick check
```

Comment:
We can create pre-allocated UniqueSet with:
make(typeutil.UniqueSet, cap)

Reply:
Map cap is just an indicator for bucket size(2^N). It might help when offline segment size is large. Normally, this offline segments would be large, and default size will be fine.